feat: :sparkles: addes x-priority option to email header and frappe.sendmail

jll-02 opened the pull request:
## Description

I want to be able to send email with different priorities / importance shown in the email, e.g. in outlook shown as an exclamation mark or an arrow. 
This should be an optional argument since I want to send normal emails most of the time and the existing code still has to work.

This PR introduces this change:
I can set the "X-Priority" when calling the `frappe.sendmail` function. This argument will be passed to the email header where it will be set to the correct Header Setting to take effect.

## Additional notes
I firstly created this with an Enum however I wasn't sure where to create the Enum. I would have had to import this very specific Enum in the `frappe/__init__.py` to set the default value. I did not feel comfortable with it. I would be willing to use an enum again if someone could tell me where best to create it and if it would be alright to import it in the `frappe/__init__.py`.

If there are other questions about my code, feel free to ask 


## Backport
when merging this a backport to v-15 would be much appreciated.

s-aga-r commented:
@jll-02 Since `X-Priority` is not universally supported by modern SMTP servers (like Stalwart) and clients, we should avoid setting a default priority of 3. Instead, the `X-Priority` header should only be included when `x_priority` is explicitly specified.

Commit:
feat: :sparkles: addes x-priority option to email header and the frappe.sendmail function

creating the option to give the email an importance.
1 = Highest, 3 = Normal, 5 = Lowest
commonly used to flag the importance of emails

jll-02 commented:
@s-aga-r Good Point.
I have changed the code so the `X-Priority` will appended to the header if the Priority is not normal. 
This will still work since there is no marking for normal mails. 

If you have anything else, please let me know
